libs/designsystem/item/src/item.component.ts

```diff
@@ -64,10 +87,20 @@ export class ItemComponent {
     }
   }
 
-  get _renderAsButton() {
+  get _renderAsButton(): boolean {
     // We shouldn't render item as a button if the item contains
     // nested interactive, i.e. checkbox, radio or toggle:
 
-    return this.selectable && !(this.checkbox || this.radio || this.toggle || this.button);
+    return this.selectable && !this._containsNestedInteractives;
+  }
+
+  get _renderAsLink(): boolean {
+    // We shouldn't render item as a link if the item contains
+    // nested interactive, i.e. checkbox, radio or toggle:
+    return this.href && !this._containsNestedInteractives;
+  }
+
+  get _containsNestedInteractives(): boolean {
```

Comment:
Nice! Consider making this a private helper function if it only called in the component class.

```suggestion
private containsNestedInteractives(): boolean {
```